src/mesh_analysis.rs

```diff
@@ -248,19 +139,20 @@ mod tests {
         let oriented_edges_with_valency_2_correct =
             vec![OrientedEdge::new(2, 0), OrientedEdge::new(0, 2)];
 
-        let oriented_edges_with_valency_1 = find_edges_with_valency(&edge_valency_map, 1);
-        let oriented_edges_with_valency_2 = find_edges_with_valency(&edge_valency_map, 2);
+        let oriented_edges_with_valency_1: Vec<_> =
+            find_edges_with_valency(&edge_valency_map, 1).collect();
+        let oriented_edges_with_valency_2: Vec<_> =
+            find_edges_with_valency(&edge_valency_map, 2).collect();
 
-        for o_e in oriented_edges_with_valency_1 {
-            assert!(oriented_edges_with_valency_1_correct
-                .iter()
-                .any(|e| *e == o_e));
+        assert_eq!(&oriented_edges_with_valency_1.len(), &cast_usize(4));
```

Comment:
Unnecessary `&` on both side of the assert, and unnecessary `cast_usize` (99% sure)